utils/components.py

```diff
@@ -61,3 +69,311 @@ def ilab_importer_op(repository: str, release: str, base_model: dsl.Output[dsl.M
+                print(f"OCI Secret has auth token present for {registry_name}")
+            elif secret.type == "kubernetes.io/dockercfg":
+                # handle authentication if secret provided is kubernetes.io/dockercfg
+                dockercfg_json = json.loads(
+                    base64.b64decode(secret.data[".dockercfg"]).decode("utf-8")
+                )
+                if registry_name not in docker_config_json.keys():
+                    print(f"""
+                    ################################################################## ERROR ################################################################################
+                    # OCI Secret {sdg_oci_docker_secret} does not have an auth token present for {registry_name}. Ensure that the secret provided has the proper auth token #
+                    #########################################################################################################################################################
+                    """)
+                    sys.exit(1)
+                print(f"""
+                ######################## INFO ###########################
+                # OCI Secret has auth token present for {registry_name} #
+                #########################################################
+                """)
+        except ApiException as e:
+            print(f"""
+            ############################################## ERROR #################################################################
+            # Secret {sdg_oci_docker_secret} does not exist. Ensure you created a secret with this name in namespace {namespace} #
+            ######################################################################################################################
+            """)
+            sys.exit(1)
+
+
+@dsl.container_component
+def test_taxonomy_repo(sdg_repo_url: str):
+    return dsl.ContainerSpec(
+        PYTHON_IMAGE,
+        ["/bin/sh", "-c"],
+        [
+            f"""
+            # Increase logging verbosity
+            set -x &&
+
+            # Add TLS Parameters if CA Cert exists and is non-zero size
+            ADDITIONAL_CLONE_PARAMS=""
+            if [ -s "$TAXONOMY_CA_CERT_PATH" ]; then
+                ADDITIONAL_CLONE_PARAMS="-c http.sslVerify=true -c http.sslCAInfo=$TAXONOMY_CA_CERT_PATH"
+            fi
+
+            # ls-remote will fail if repo is not valid
+            git ls-remote {sdg_repo_url} > /dev/null;
+            """
+        ],
+    )
+
+
+@dsl.pipeline(display_name="Pre-requisite check")
+def pre_requisites_check_op(
+    sdg_repo_url: str,
+    sdg_base_model: str,
+    sdg_oci_secret: str,
+    judge_cm_name: str,
+    judge_secret_name: str,
+    teacher_cm_name: str,
+    teacher_secret_name: str,
+    model_registry_endpoint: str,
+):
+    """
+    Pre-validation checks for the InstructLab pipeline.
+    """
+    import os
+
+    # TODO: Is it required to disable caching for all validation steps?
```

Comment:
I think the answer is yes. Did you find something different?

Reply:
No, we should disable caching as the subsequent run with same parameters will skip validation. Removing this TODO.